sphinx/themes/basic/static/searchtools.js

```diff
@@ -304,10 +309,13 @@ const Search = {
       if (title.toLowerCase().trim().includes(queryLower) && (queryLower.length >= title.length/2)) {
         for (const [file, id] of foundTitles) {
           let score = Math.round(100 * queryLower.length / title.length)
+          let anchorIsDocumentTitle = titles[file] === title
```

Comment:
Could we pre-calculate this when building the `alltitles` in Python, so that the client doesn't have to transform each entry every time?

Reply:
This appears theoretically possible.  I've drafted some sample code to illustrate that alternative here: https://github.com/sphinx-doc/sphinx/pull/12047 - however it does adjust the contents of the index, and that could be considered a disadvantage.